packages/forms/src/index.tsx

```diff
@@ -113,6 +118,59 @@ const inputTagProps = <T extends InputTagProps>(
   return tagProps
 }
 
+// A hook-like function merge field props and return props (ref, onChange and onBlur primarily) of RHF v7 register()
+const useFieldRegister = <
+  T extends Omit<ValidatableFieldProps, 'defaultValue'>,
+  E
+>(
+  props: T,
+  ref?: React.ForwardedRef<E>
+) => {
+  const { register } = useFormContext()
+  const validation = props.validation || { required: false }
+
+  // Primarily for TextAreaField
+  if (!validation.validate && props.transformValue === 'Json') {
+    validation.validate = jsonValidation
+  }
+
+  const tagProps = inputTagProps(props) as T & {
+    onChange?: React.ChangeEventHandler<T>
+    onBlur?: React.FocusEventHandler<T>
+  }
```

Comment:
I'm not happy about this type casting but only have few hours today, open for any better suggestions